src/core/service.c

```diff
@@ -2106,13 +2090,19 @@ static void service_enter_dead(Service *s, ServiceResult f, bool allow_restart)
                         return service_enter_dead(s, SERVICE_FAILURE_RESOURCES, /* allow_restart= */ false);
                 }
 
-                log_unit_debug(UNIT(s), "Next restart interval calculated as: %s", FORMAT_TIMESPAN(restart_usec_next, 0));
-
                 /* If the relevant option is set, and the unit doesn't already have logging level set to
                  * debug, enable it now. Make sure to overwrite the state in /run/systemd/units/ too, to
                  * ensure journald doesn't prune the messages. The previous state is saved and restored
                  * once the auto-restart flow ends. */
-                service_set_debug_invocation(s, /* enable= */ true);
+                if (s->restart_mode == SERVICE_RESTART_MODE_DEBUG) {
```

Comment:
https://github.com/systemd/systemd/pull/33877#discussion_r1728815318 explicitly asked to remove the conditionalization, which is being added back here

Reply:
I think conditionalize here makes more sense, also corresponding to the `RESTART_MODE_DIRECT` branch above.